ci/run_clang_tidy.sh

```diff
@@ -113,15 +115,22 @@ elif [[ "${RUN_FULL_CLANG_TIDY}" == 1 ]]; then
   echo "Running a full clang-tidy"
   run_clang_tidy
 else
-  if [[ -z "${DIFF_REF}" ]]; then
-    if [[ "${BUILD_REASON}" == *CI ]]; then
-      DIFF_REF="HEAD^"
-    else
-      DIFF_REF=$("${ENVOY_SRCDIR}"/tools/git/last_github_commit.sh)
+    if [[ -z "${DIFF_REF}" ]]; then
+        # Postsubmit
+        if [[ "${BUILD_REASON}" == *CI ]]; then
+            DIFF_REF="HEAD^"
+        # Presubmit/PR
+        elif [[ -n "${BUILD_REASON}" ]]; then
+            # Common ancestor commit
+            git fetch origin
+            git fetch --unshallow
+            DIFF_REF="$(git merge-base HEAD "${DIFF_TARGET_BRANCH}")"
+        else
+            DIFF_REF=$("${ENVOY_SRCDIR}"/tools/git/last_github_commit.sh)
```

Comment:
I guess this applies to running it locally (no pre/post submit-CI).
Will it make sense to have this the same as presubmit-PR sha (so a dev can reproduce the same behavior as the when submitting a PR)?

Reply:
this is only set up for PRs - we probably want it to work similar for local - but then we need to make things like the `upstream` (`origin` in this PR) configurable

Reply:
Ok, let's add a TODO here, referencing the current PR so this can be tracked later. WDYT?

Reply:
yep +1 - ill add a comment now - i think the follow up is to resolve this here _and_ to update/add docs about it somewhere

not sure why it takes so long to run this, and also not entirely clear why its necessary to compile everything to run a static analyzer - but either way, as it stands, devs will want to start running this to avoid CI fails

at least once it is run once, and using just the clang-tidy-diff it can run in a ~reasonable time